script/base/DeployMorphoTokenBase.sol

```diff
@@ -0,0 +1,46 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+pragma solidity ^0.8.0;
+
+import "../../lib/forge-std/src/Script.sol";
+import "../../lib/forge-std/src/console.sol";
+
+import {MorphoTokenOptimism} from "../../src/MorphoTokenOptimism.sol";
+import {ERC1967Proxy} from
+    "../../lib/openzeppelin-contracts-upgradeable/lib/openzeppelin-contracts/contracts/proxy/ERC1967/ERC1967Proxy.sol";
+
+contract DeployMorphoTokenBase is Script {
+    address public constant MORPHO_DAO = 0xcBa28b38103307Ec8dA98377ffF9816C164f9AFa;
+    address public REMOTE_TOKEN;
+    address public constant BRIDGE = 0x4200000000000000000000000000000000000010;
+
+    bytes32 public IMPLEMENTATION_SALT;
+    bytes32 public PROXY_SALT;
+
+    address public implementationAddress;
+    MorphoTokenOptimism public token;
+    address public wrapperAddress;
+    address public newMorphoAddress;
+
+    function run() public returns (address) {
+        vm.createSelectFork(vm.rpcUrl("base"));
+
+        vm.startBroadcast();
+
+        // Deploy Token implementation
+        implementationAddress = address(new MorphoTokenOptimism{salt: IMPLEMENTATION_SALT}(REMOTE_TOKEN, BRIDGE));
+
+        console.log("Deployed token implementation at", implementationAddress);
+
+        // Deploy Token proxy
+        token = MorphoTokenOptimism(address(new ERC1967Proxy{salt: PROXY_SALT}(implementationAddress, hex"")));
```

Comment:
can we initialize the token at construction of the proxy pls? Will avoid highjacking the implem.

Reply:
it's done in the contractor

https://github.com/morpho-org/morpho-token-upgradeable/blob/04cf118ffa65aa14fe55e1bd8d63df49e19b7e73/src/DelegationToken.sol#L67